### PR TITLE
Added minimum and maximum implementation for Loc

### PR DESCRIPTION
**Pull request by giovannitangredi**
This PR adds minimum and maximum value for the Loc metric.

Given the  high number of fields the Sloc, Ploc, Cloc and Lloc substructs have been created to make the code more readable.
Partly resolving Issue https://github.com/mozilla/rust-code-analysis/issues/410 .

### CODE REVIEW

File: rust-code-analysis-web/src/main.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![recursion_limit = "256"]
```
Comment (calixteman) [inline rust-code-analysis-web/src/main.rs:1]: Why this change ?
It requires at least a good comment to explain the issue it's supposed to solve.

Comment (giovannitangredi) [reply rust-code-analysis-web/src/main.rs:1]: After I added the new files for the json in server.rs tests an error arose in compile time telling that the recursion limit was reached, this change was the most fast  and  straightforward way to resolve the issue.
here there is the error for more context:
```
error: recursion limit reached while expanding `json_internal!`
   --> rust-code-analysis-web/src/web/server.rs:630:24
    |
630 |           let expected = json!({
    |  ________________________^
631 | |             "id": "1234",
632 | |             "language": "python",
633 | |             "spaces": {"kind": "unit",
...   |
687 | |                                    "spaces": []}]}
688 | |         });
    | |__________^
    |
    = help: consider increasing the recursion limit by adding a `#![recursion_limit = "256"]` attribute to your crate (`rust_code_analysis_web`)
    = note: this error originates in the macro `json_internal` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Comment (giovannitangredi) [reply rust-code-analysis-web/src/main.rs:4]: Added a comment also in the code

**Review by calixteman — APPROVED** [review]
Overall it looks good to me.
Thank you for doing this.
